food-waste-express-api/routes/home.js

```diff
@@ -134,15 +138,35 @@ router.post('/:objectId', async (req, res) => {
     
     try {
         let userBasket = await getUserBasket(currentUserId) //returns first instance
+        let tempBasket = []
+
         if (userBasket != null) {
-            userBasket.addUnique("basketOfProductId", {productId, quantity})
-            await userBasket.save()
+            let doesNotHaveDuplicate = true;
+            if (userBasket.get("basketOfProducts").length > 0) {
+                for (productDetail of userBasket.get("basketOfProducts")) {
+                    if (productDetail.productId == productId) {
+                        let newQuantity = productDetail.quantity + quantity
+                        tempBasket.push({ "productId": productId, "quantity": newQuantity.toString() })
+                        doesNotHaveDuplicate = false;
+                    }
+                    else {
+                        tempBasket.push(productDetail)
+                    }
+                }
+                userBasket.removeAll("basketOfProducts", { productId, quantity })
+                userBasket.set("basketOfProducts", tempBasket)
+                await userBasket.save()
+            }
+            if (doesNotHaveDuplicate) {
+                userBasket.addUnique("basketOfProducts", { productId, quantity })
+                await userBasket.save()
+            }
         }
         else {
             const Basket = Parse.Object.extend("Basket");
             let basket = new Basket();
             basket.set("userId", currentUserId);
-            basket.set("basketOfProductId", [{productId, quantity}])
+            basket.set("basketOfProducts", [{productId, quantity}])
```

Comment:
ditto, change this into a function 'basket.setUserID(int currentUserID)' and 'basket.addNewProducts([int productID, int quantity])'